### PR TITLE
feat(deps): update deps, support of eslint 8 (rebase)

### PR DESCRIPTION
**Pull request by Delagen**

### CODE REVIEW

**Comment by magyargergo** [comment]
Hello @Delagen, 
May I ask when we can expect this PR?

**Comment by gund** [comment]
Hey @Delagen, thanks for the PR!

**Review by gund — APPROVED** [review]

File: src/rules/deprecation.ts [hunk 1]
```diff
@@ -132,7 +131,7 @@ function isDeclaration(
       // Prevent considering initializer, extends, or implements to be declaration
       return parent.id === id;
 
-    case 'ClassProperty':
+    case 'PropertyDefinition':
```
Comment (jdom) [inline src/rules/deprecation.ts:134]: this change is backwards compatible with eslint 7?

Comment (gund) [reply src/rules/deprecation.ts:134]: Hmm, good question.
Now that I look at it I believe it's not and will only work with eslint 8+...

Comment (gund) [reply src/rules/deprecation.ts:134]: Should we just keep both cases instead of replacing the old one?

Comment (gund) [reply src/rules/deprecation.ts:134]: This was fixed in #41 and released in v1.3.2!

**Pull request merged**